app/src/main/java/com/github/libretube/api/StreamsExtractor.kt

```diff
@@ -58,13 +55,71 @@ fun StreamInfoItem.toStreamItem(
 )
 
 object StreamsExtractor {
-    suspend fun extractStreams(videoId: String): Streams {
+    private val extractorScope = CoroutineScope(Dispatchers.Default + SupervisorJob())
+
+  suspend inline fun extractStreams(videoId: String): Streams = withContext(Dispatchers.IO) {
         if (!PlayerHelper.disablePipedProxy || !PlayerHelper.localStreamExtraction) {
-            return RetrofitInstance.api.getStreams(videoId)
+            return@withContext RetrofitInstance.api.getStreams(videoId)
+        }
+
+        val resp = StreamInfo.getInfo("$YOUTUBE_FRONTEND_URL/watch?v=$videoId")
+        val deferredOperations = coroutineScope {
+            val dislikes = async {
+                if (PlayerHelper.localRYD) runCatching {
+                    RetrofitInstance.externalApi.getVotes(videoId).dislikes
+                }.getOrElse { -1 } else -1
+            }
+
+            val relatedStreams = async {
+                resp.relatedItems
+                    .filterIsInstance<StreamInfoItem>()
+                    .chunked(4)
+                    .map { chunk ->
+                        chunk.map { item ->
+                            async { item.toStreamItem() }
```

Comment:
I don't think that this specific part needs to be parallelized, `toStreamItem` doesn't do any API calls.